src/ds/d400/d400-factory.cpp

```diff
@@ -1356,6 +1233,15 @@ namespace librealsense
         return matcher_factory::create(RS2_MATCHER_DEFAULT, streams);
     }
 
+    std::shared_ptr<matcher> rs436_device::create_matcher(const frame_holder& frame) const
+    {
+        std::vector<stream_interface*> streams = { _depth_stream.get() , _left_ir_stream.get() , _right_ir_stream.get(), _color_stream.get() };
```

Comment:
Why do we need all this code duplication?
Can rs436/435i/430i_device etc call a common "create_mm_streams" function?